### PR TITLE
Add build skeleton for Q subplugin

### PR DESCRIPTION
**Pull request by rli**
This commit sets up the `:plugin-amazonq:...` structure and allows `:plugin-amazonq:runIde` and related plugin tasks to perform as expected.

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by andrewyuq — APPROVED** [review]

File: plugins/amazonq/codewhisperer/jetbrains-community/resources/META-INF/plugin-codewhisperer.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<!-- Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved. -->
+<!-- SPDX-License-Identifier: Apache-2.0 -->
+
+<idea-plugin>
+    <actions>
+        <action class="Hello" id="q.hello" />
```
Comment (manodnyab) [inline plugins/amazonq/codewhisperer/jetbrains-community/resources/META-INF/plugin-codewhisperer.xml:6]: Can be deleted?

File: plugins/amazonq/codewhisperer/jetbrains-community/src/Hello.kt [hunk 2]
```diff
@@ -0,0 +1,15 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.project.DumbAware
+import com.intellij.openapi.ui.MessageDialogBuilder
+import software.aws.toolkits.resources.AmazonQBundle
+
+class Hello : AnAction(AmazonQBundle.message("q.hello")), DumbAware {
```
Comment (manodnyab) [inline plugins/amazonq/codewhisperer/jetbrains-community/src/Hello.kt:10]: Test class, can be deleted?

Comment (rli) [reply plugins/amazonq/codewhisperer/jetbrains-community/src/Hello.kt:10]: whoever moves the code can delete it.
it's a sample to show what we're expecting

File: plugins/amazonq/codewhisperer/jetbrains-ultimate/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+plugins {
+    id("toolkit-intellij-subplugin")
+}
+
+intellijToolkit {
+    ideFlavor.set(IdeFlavor.IU)
+}
+
+dependencies {
+    compileOnly(project(":plugin-amazonq:codewhisperer:jetbrains-community"))
```
Comment (manodnyab) [inline plugins/amazonq/codewhisperer/jetbrains-ultimate/build.gradle.kts:15]: Should this also take in shared:jb-community or will it be transitive?

Comment (rli) [reply plugins/amazonq/codewhisperer/jetbrains-ultimate/build.gradle.kts:15]: transitive

**Review by manodnyab — APPROVED** [review]

**Pull request merged**